student/views.py

```diff
@@ -1,3 +1,25 @@
-# from django.shortcuts import render
+from django.contrib.auth.mixins import UserPassesTestMixin
+from django.shortcuts import render
+from .models import Student
+from django.views.generic.edit import UpdateView
 
-# Create your views here.
+
+def account_update_success(request):
+    is_student = Student.is_student(request.user.id)
+    return render(request, 'account_settings_update_success.html', {"is_student": is_student})
+
+
+class UpdateStudentSettings(UserPassesTestMixin, UpdateView):
```

Comment:
Do not use class based views:
https://spookylukey.github.io/django-views-the-right-way/

It generally makes the code harder to understand while providing very little benefit.

Reply:
@ifireball 
Um, ok, I'll try to recreate it using view functions..

Reply:
@ifireball 
Done.